2-Draft-Accepted/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,239 @@
+
+### Specification for `ConvertFrom-Markdown`
+
+```PowerShell
+
+ConvertFrom-Markdown [-Path] <string[]> [-AsVT100EncodedString] [<CommonParameters>]
+
+ConvertFrom-Markdown [-LiteralPath] <string[]> [-AsVT100EncodedString] [<CommonParameters>]
+
+ConvertFrom-Markdown [-InputObject] <psobject> [-AsVT100EncodedString] [<CommonParameters>]
+
+```
+
+#### Parameters
+
+- **Path** : Accepts an array of file paths with markdown content.
+- **LiteralPath** : Accepts an array of literal paths with markdown content.
+- **InputObject** : Accepts an InputObject of `System.IO.FileInfo`, `string` type.
+- **AsVT100EncodedString** : When selected, the `VT100EncodedString` property is populated.
+
+There will be support for changing the colors for various elements using the `Set-MarkdownOption` cmdlet. To retrieve the current settings `Get-MarkdownOption` cmdlet can be used.
+
+#### Output Type
+
+The output type will be `MarkdownInfo` object with properties for `Html`, `VT100EncodedString` and `Tokens`.
+
+### Specification for `Set-MarkdownOption`
+
+```PowerShell
+
+Set-MarkdownOption [-Header1Color <string>] [-Header2Color <string>] [-Header3Color <string>] [-Header4Color <string>] [-Header5Color <string>] [-Header6Color <string>] [-CodeBlockForegroundColor <string>] [-CodeBlockBackgroundColor <string>] [-ImageAltTextForegroundColor <string>] [-LinkForegroundColor <string>] [-ItalicsForegroundColor <string>] [<CommonParameters>]
+
+Set-MarkdownOption -Theme <string> [<CommonParameters>]
+
+Set-MarkdownOption -InputObject <psobject> [<CommonParameters>]
+
+```
+
+The properties can individually customize the rendering on console.
+The properties for color must be expressed as a `VT100` escape sequence, like
+
+```PowerShell
+
+Set-MarkdownOption -Header1Color "$([char]0x1b)[7m"
+
+```
+
+Dark will be the default theme.
+The individual colors for the dark theme are specified in the subsequent sections.
+
+### Specification for `Get-MarkdownOption`
+
+```PowerShell
+
+Get-MarkdownOption [<CommonParameters>]
+
+```
+
+### Specification for `Export-MarkdownOption`
+
+```PowerShell
+
+Export-MarkdownOption -Path <string> [<CommonParameters>]
+
+Export-MarkdownOption -LiteralPath <string> [<CommonParameters>]
+
+```
+
+Export the current markdown settings to a JSON file.
+
+### Specification for `Import-MarkdownOption`
+
+```PowerShell
+
+Import-MarkdownOption -Path <string> [<CommonParameters>]
+
+Import-MarkdownOption -LiteralPath <string> [<CommonParameters>]
+
+```
+
+Import the markdown settings from the specified file and returns a PSObject of the options.
+This can be used as an `InputObject` for `Set-MarkdownOption`.
+
+### Specification for `Show-Markdown`
+
+```Powershell
+
+Show-Markdown -InputObject <psobject[]> [<CommonParameters>]
+
+Show-Markdown -InputObject <psobject[]> -UseBrowser [<CommonParameters>]
+
+```
+
+Render the `VT100EncodedString` property of `MarkdownInfo` on console.
```

Comment:
Related my previous comment - What about `Out-Default`, `Out-Mardown` and ETS?

Reply:
With a default formatter, `Out-Default` will work.